File: Game_Of_Life_Python/Organisms/Plants/guarana.py
from Organisms.organism import OrganismInitialData
from Organisms.plant import Plant


class Guarana(Plant):

    static_counter = 0

    def __init__(self, name, row, column, given_strength = -1, given_initiative = -1):
        strength = OrganismInitialData[name]["strength"]
        initiative = OrganismInitialData[name]["initiative"]
        character = OrganismInitialData[name]["character"]
        if given_strength != -1: strength = given_strength
        if given_initiative != -1: initiative = given_initiative
        super().__init__(strength, initiative, name, character, row, column, "guarana.png")
        Guarana.static_counter += 1
        self.type = Guarana

    def action(self, grid_board):
        return self.default_plant_action(grid_board, Guarana, "Guarana")

    def collision(self, grid_board, organisms, current_index):
        pass

    def get_static_counter(self):
        return Guarana.static_counter

File: GameOfLife_Cpp/source/src/guarana.cpp
#include "guarana.hpp"

int Guarana::GUARANA_STATIC_COUNTER = 0;

Guarana::Guarana(int row, int column)
    : Plant(0, 0, 0, "Guarana", 'U', row, column, "guarana.png", OrganismType::GUARANA)
{
    GUARANA_STATIC_COUNTER++;
}

Guarana::Guarana(int row, int column, int strength, int initiative, int age)
    : Plant(strength, initiative, age, "Guarana", 'U', row, column, "guarana.png", OrganismType::GUARANA)
{
    GUARANA_STATIC_COUNTER++;
}

Guarana::~Guarana()
{
    GUARANA_STATIC_COUNTER--;
}

ActionResult Guarana::action(std::vector<std::vector<char>>& grid_board)
{
    return this->default_action_plant(grid_board);
}

CollisionResult Guarana::collision(std::vector<std::vector<char>>& grid_board, std::vector<Organism*>& organisms, int current_index)
{
    return this->default_collision_plant();
}

int Guarana::get_static_counter()
{
    return GUARANA_STATIC_COUNTER;
}